packages/react/src/auto/polaris/inputs/PolarisDateTimePicker.tsx

```diff
@@ -0,0 +1,132 @@
+import type { DatePickerProps, TextFieldProps } from "@shopify/polaris";
+import { DatePicker, Icon, InlineStack, Popover, TextField } from "@shopify/polaris";
+import { CalendarIcon } from "@shopify/polaris-icons";
+import { format } from "date-fns";
+import { utcToZonedTime, zonedTimeToUtc } from "date-fns-tz";
+import React, { useCallback, useMemo, useState } from "react";
+import { useController } from "react-hook-form";
+import { useFieldMetadata } from "../../hooks/useFieldMetadata.js";
+import type { DateTimeState } from "./PolarisTimePicker.js";
+import PolarisTimePicker from "./PolarisTimePicker.js";
+
+export const copyTime = (to: Date, input: Date) => {
+  to.setHours(input.getHours());
+  to.setMinutes(input.getMinutes());
+  to.setSeconds(input.getSeconds());
+  to.setMilliseconds(input.getMilliseconds());
+};
+
+export const getDateTimeObjectFromDate = (dateTime: Date) => {
+  return {
+    month: dateTime.getMonth(),
+    year: dateTime.getFullYear(),
+    day: dateTime.getDay(),
+    hour: dateTime.getHours() > 12 ? (dateTime.getHours() - 12).toString() : dateTime.getHours().toString(),
+    minute: dateTime.getMinutes().toString().padStart(2, "0"),
+    ampm: dateTime.getHours() >= 12 ? "PM" : "AM",
+  };
+};
+
+export const getDateFromDateTimeObject = (dateTime: DateTimeState) => {
+  const date = new Date();
+  date.setMonth(dateTime.month);
+  date.setFullYear(dateTime.year);
+  date.setDate(dateTime.day);
+  date.setHours(dateTime.ampm === "PM" ? parseInt(dateTime.hour) + 12 : parseInt(dateTime.hour));
+  date.setMinutes(parseInt(dateTime.minute));
+  date.setSeconds(0);
+  date.setMilliseconds(0);
+  return date;
+};
+
+export const PolarisDateTimePicker = (props: {
+  field: string;
+  id?: string;
+  value?: Date;
+  timeLabel?: string;
```

Comment:
I don't think this is needed since it can already be given from `props.timePickerProps?.label`